### PR TITLE
Added HunSum2 dataset with results

### PR DESCRIPTION
**Pull request by x-tabdeveloping**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

Added the first Hungarian Retrieval dataset to the benchmark.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] Run tests locally to make sure nothing is broken using `make test`.
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [POINTS.md](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) file.

### CODE REVIEW

**Comment by x-tabdeveloping** [comment]
Addresses #384, but I decided to use HunSum2 instead as it is an updated version of the same dataset.

**Comment by KennethEnevoldsen** [comment]
> I could not do this, as my computer doesn't have enough memory, it crashed multiple times trying to run the tests, everything I could run passes

Does it also crash if you only use `pytest` or `pytest -n 4`

File: mteb/tasks/Retrieval/hun/HunSum2.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        description=(
+            "HunSum-2-abstractive is a Hungarian dataset containing news articles along with lead, titles and metadata."
+        ),
+        reference="https://arxiv.org/abs/2404.03555",
+        type="Retrieval",
+        category="s2p",
+        eval_splits=["test"],
+        eval_langs=["hun-Latn"],
+        main_score="ndcg_at_10",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/hun/HunSum2.py:25]:
```suggestion
        main_score="ndcg_at_1",
```
Given the very high scores I would lower it to 1.

File: mteb/tasks/Retrieval/hun/HunSum2.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        date=(
+            "1848-12-15",
+            "2024-03-19",
+        ),
+        form=["written"],
+        domains=["News"],
+        task_subtypes=["Article retrieval"],
+        license="CC-BY 4.0",
+        socioeconomic_status="mixed",
+        annotations_creators="human-annotated",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/hun/HunSum2.py:35]: Isn't this derived?

**Review by KennethEnevoldsen — APPROVED** [review]
This looks good. A few minor comments

File: mteb/tasks/Retrieval/hun/HunSum2.py (hunk 2 above)
Comment (x-tabdeveloping) [reply mteb/tasks/Retrieval/hun/HunSum2.py:35]: Yeah, you're right sorry about that

File: mteb/tasks/Retrieval/hun/HunSum2.py (hunk 1 above)
Comment (x-tabdeveloping) [reply mteb/tasks/Retrieval/hun/HunSum2.py:25]: Okay will change

**Comment by KennethEnevoldsen** [comment]
Setting this to auto-merge!

**Pull request merged**